scenarioo-client/app/build/scenarios-overview/scenarios-overview.component.ts

```diff
@@ -0,0 +1,239 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Component, HostListener, OnInit} from '@angular/core';
+import {SelectedBranchAndBuildService} from '../../shared/navigation/selectedBranchAndBuild.service';
+import {BranchesAndBuildsService} from '../../shared/navigation/branchesAndBuilds.service';
+import {ScenarioResource} from '../../shared/services/scenarioResource.service';
+import {LabelConfigurationMap, LabelConfigurationsResource} from '../../shared/services/labelConfigurationsResource.service';
+import {SelectedComparison} from '../../diffViewer/selectedComparison.service';
+import {LocationService} from '../../shared/location.service';
+import {ILabelConfiguration, IScenario, IScenarioSummary, IUseCaseScenarios} from '../../generated-types/backend-types';
+import {ConfigurationService} from '../../services/configuration.service';
+import {downgradeComponent} from '@angular/upgrade/static';
+import {OrderPipe} from 'ngx-order-pipe';
+import {forkJoin} from 'rxjs';
+import {UseCaseDiffInfoService} from '../../diffViewer/services/use-case-diff-info.service';
+import {ScenarioDiffInfosService} from '../../diffViewer/services/scenario-diff-infos.service';
+import {DiffInfoService} from '../../diffViewer/diffInfo.service';
+import {MetadataTreeCreatorPipe} from '../../pipes/metadataTreeCreator.pipe';
+import {RelatedIssueResource, RelatedIssueSummary} from '../../shared/services/relatedIssueResource.service';
+import {RouteParamsService} from '../../shared/route-params.service';
+import {MetadataTreeListCreatorPipe} from '../../pipes/metadataTreeListCreator.pipe';
+import {FilterPipe} from '../../pipes/filter.pipe';
+
+@Component({
+    selector: 'sc-scenarios-overview',
+    template: require('./scenarios-overview.component.html'),
+    styles: [require('./scenarios-overview.component.css').toString()],
+})
+
+export class ScenariosOverviewComponent implements OnInit {
+
+    useCaseName: string;
+
+    scenarios: IScenarioSummary[] = [];
+    scenario: IScenario[] = [];
+
+    searchTerm: string;
+
+    order: string = 'name';
+    sortedScenarios: any[];
+    reverse: boolean = false;
+
+    arrowkeyLocation: number = 0;
+
+    labelConfigurations: LabelConfigurationMap = undefined;
+    labelConfig: ILabelConfiguration = undefined;
+
+    getStatusStyleClass = undefined;
+    comparisonExisting = undefined;
```

Comment:
In AngularJS it was ok to define functions/methods as variables. In Angular this needs not to be done. Remove this reference and replace it with a method.